src/iceberg/iceberg_parser/iceberg_parser.py

```diff
@@ -0,0 +1,234 @@
+"""
+Author: Ioannis Paraskevakos
+License: MIT
+Copyright: 2018-2019
+"""
+
+from __future__ import print_function
+import argparse
+import sys
+from collections import defaultdict
+
+
+class IcebergParser(object):
+    """
+    This class is the argument parser of the ICEBERG software tool.
+    """
+
+    def __init__(self):
+        """
+        The constructor
+        """
+        parser = argparse.ArgumentParser(description='ICEBERG command description:',
+                                         epilog='''Enjoy our tool!''')
+        required_args = parser.add_argument_group()
+        required_args.title = 'Required Arguments'
+        required_args.add_argument('--resource', '-r', help='Where the execution will happen',
+                                   type=str)
+        required_args.add_argument('--queue', '-q', help='The queue of the resource',
+                                   type=str)
+        required_args.add_argument('--cpus', '-c', help='How many CPUs will be required',
+                                   type=int)
+        required_args.add_argument('--gpus', '-g', help='How many GPUs will be required',
+                                   type=int)
+        required_args.add_argument('--input_path', '-ip', help='Where the input images are',
+                                   type=str)
+        required_args.add_argument('--output_path', '-op', help='Where the results should be saved',
+                                   type=str)
+        required_args.add_argument('--walltime', '-w', help='The estimated execution time',
+                                   type=int)
+        required_args.add_argument('--analysis', '-a', help='The type of analysis to be executed',
+                                   type=str)
+
+        if 'python' in sys.argv[0]:
+            argvs = sys.argv[2:]
+        else:
+            argvs = sys.argv[1:]
+        actual_argvs = list()
+        while argvs:
+            argv = argvs.pop(0)
+            if '=' in argv:
+                actual_argvs.append(argv)
+            elif '--' in argv and not '=' in argv and argv not in ['--help','-h']:
+                temp_argv = argv
+                argv = argvs.pop(0)
```

Comment:
Hello @iparask  : 
i was trying to do the following : 
`iceberg --resource xsede.bridges --queue=RM --cpus 28 --gpus 2 --input_path=/home/aymen/ --output_path=/home/aymen/ --walltime 90 --analysis` and i did miss the --analysis on purpose ,

 it should say `few arguments `  instead i got the following :

`File "/home/aymen/miniconda3/envs/conda_env1/bin/iceberg", line 12, in <module>
    IcebergParser()
  File "/home/aymen/miniconda3/envs/conda_env1/lib/python2.7/site-packages/iceberg/iceberg_parser/iceberg_parser.py", line 54, in __init__
    argv = argvs.pop(0)
IndexError: pop from empty list `